src/TorchVision/AutoAugment.cs

```diff
@@ -0,0 +1,162 @@
+                                IList<float>? fill)
+        {
+            if (op_name == "ShearX") {
+                img = F.affine(
+                    img,
+                    angle: 0.0f,
+                    translate: new[] { 0, 0 },
+                    scale: 1.0f,
+                    shear: new[] { (float)((180.0f / Math.PI) * Math.Atan(magnitude)), 0.0f },
+                    interpolation: interpolation,
+                    fill: fill?.FirstOrDefault());
+            } else if (op_name == "ShearY") {
+                img = F.affine(
+                    img,
+                    angle: 0.0f,
+                    translate: new[] { 0, 0 },
+                    scale: 1.0f,
+                    shear: new[] { 0.0f, (float)((180.0f / Math.PI) * Math.Atan(magnitude)) },
+                    interpolation: interpolation,
+                    fill: fill?.FirstOrDefault());
+            } else if (op_name == "TranslateX") {
+                img = F.affine(
+                    img,
+                    angle: 0.0f,
+                    translate: new[] { (int)(magnitude), 0 },
+                    scale: 1.0f,
+                    interpolation: interpolation,
+                    shear: new[] { 0.0f, 0.0f },
+                    fill: fill?.FirstOrDefault());
+            } else if (op_name == "TranslateY") {
+                img = F.affine(
+                    img,
+                    angle: 0.0f,
+                    translate: new[] { 0, (int)(magnitude) },
+                    scale: 1.0f,
+                    interpolation: interpolation,
+                    shear: new[] { 0.0f, 0.0f },
+                    fill: fill?.FirstOrDefault());
+            } else if (op_name == "Rotate") {
+                img = F.rotate(img, magnitude, interpolation, fill: fill);
+            } else if (op_name == "Brightness") {
+                img = F.adjust_brightness(img, 1.0 + magnitude);
+            } else if (op_name == "Color") {
+                img = F.adjust_saturation(img, 1.0 + magnitude);
+            } else if (op_name == "Contrast") {
+                img = F.adjust_contrast(img, 1.0 + magnitude);
+            } else if (op_name == "Sharpness") {
+                img = F.adjust_sharpness(img, 1.0 + magnitude);
+            } else if (op_name == "Posterize") {
+                img = F.posterize(img, (int)magnitude);
+            } else if (op_name == "Solarize") {
+                img = F.solarize(img, magnitude);
+            } else if (op_name == "AutoContrast") {
+                img = F.autocontrast(img);
+            } else if (op_name == "Equalize") {
+                img = F.equalize(img);
+            } else if (op_name == "Invert") {
+                img = F.invert(img);
+            } else if (op_name == "Identity") {
+                // Pass
+            } else {
+                throw new ArgumentException($"The provided operator {op_name} is not recognized.");
```

Comment:
With an enum, you won't have to throw. Just sayin'... :-)

Reply:
You don't need to convince me. I much prefer enums, or better still class based dispatch. 
the idea was to copy RandAugment, then update it: https://pytorch.org/vision/main/_modules/torchvision/transforms/autoaugment.html#_apply_op